pkg/apis/kops/validation/instancegroup.go

```diff
@@ -114,6 +115,10 @@ func ValidateInstanceGroup(g *kops.InstanceGroup) field.ErrorList {
 		allErrs = append(allErrs, validateRollingUpdate(g.Spec.RollingUpdate, field.NewPath("spec", "rollingUpdate"), g.Spec.Role == kops.InstanceGroupRoleMaster)...)
 	}
 
+	if awsCloud, ok := cloud.(awsup.AWSCloud); ok {
+		allErrs = append(allErrs, awsValidateInstanceGroup(g, awsCloud)...)
```

Comment:
It would be better if `awsValidateInstanceGroup` were called when `cloud` is `nil` but the cloud provider is AWS.

Reply:
I'm not sure I understand. I can change this conditional to look at the cloud provider being AWS, but are you suggesting that `awsValidateInstanceGroup` should take a `fi.Cloud` instead of `awsup.AWSCloud`? Where should the `fi.Cloud` -> `awsup.AWSCloud` type assertion happen then?

Reply:
If the type assertion fails then (if cloud provider is AWS) it should call `awsValidateInstanceGroup` with a nil `awsup.AWSCloud`.